massa-module-cache/src/hd_cache.rs

```diff
@@ -0,0 +1,301 @@
+use crate::types::{
+    ModuleInfo, ModuleMetadata, ModuleMetadataDeserializer, ModuleMetadataSerializer,
+};
+use massa_hash::Hash;
+use massa_sc_runtime::{GasCosts, RuntimeModule};
+use massa_serialization::{DeserializeError, Deserializer, Serializer};
+use rand::RngCore;
+use rocksdb::{Direction, IteratorMode, WriteBatch, DB};
+use std::path::PathBuf;
+
+const OPEN_ERROR: &str = "critical: rocksdb open operation failed";
+const CRUD_ERROR: &str = "critical: rocksdb crud operation failed";
+const DATA_SER_ERROR: &str = "critical: metadata serialization failed";
+const DATA_DESER_ERROR: &str = "critical: metadata deserialization failed";
+const MOD_SER_ERROR: &str = "critical: module serialization failed";
+const MOD_DESER_ERROR: &str = "critical: module deserialization failed";
+const MODULE_IDENT: u8 = 0u8;
+const DATA_IDENT: u8 = 1u8;
+
+/// Module key formatting macro
+#[macro_export]
```

Comment:
I think the key formating in the ledger was refactored. Maybe we can unify the keys formatting at some point.

Reply:
I actually disagree on how the key formatting was done in the ledger, I was not able to review the PR because I was sick at that time. Furthermore we found behavioural issues when going over the testnet 19 to 20 diff changes if I remember correctly. In any case I agree with you that we need it to be unified, but I think we need to define what we'd like to do and why (which was not done for the ledger implementation). I consider this to be out of this PR scope.

Reply:
Ok, I was asking because the network restart follow-up will almost certainly need to implement something similar, so I'll be sure to include you in the design discussion ;)